### PR TITLE
Support workspace tokens in API

### PR DESCRIPTION
**Pull request by LizaShak**
# PR for issue #983

## What is being addressed

This change allows the api to authenticate workspaces app registrations tokens.

## How is this addressed

- Apply on workspaces specific routes - meaning routes that require workspace_id as part of the url
- When token issued for workspace is received
- Take the aud claim from the token
- Query the cosmos resources collection with the workspaceId==workspace_id on the request url and authorize the call only if appId param equals the aud claim.

### CODE REVIEW

**Review by TessFerrandez — APPROVED** [review]
LGTM! Haven't tested the changes, but assuming they work fine, the code looks good

File: e2e_tests/conftest.py [hunk 1]
```diff
@@ -33,3 +33,18 @@ async def token(verify) -> str:
...
+@pytest.fixture
+async def workspace_owner_token(verify) -> str:
+    async with AsyncClient(verify=verify) as client:
+
+        headers = {'Content-Type': "application/x-www-form-urlencoded"}
+        payload = f"grant_type=password&resource={config.AUTH_APP_CLIENT_ID}&username={config.USERNAME}&password={config.PASSWORD}&scope=open_id+profile+email&client_id={config.AUTH_APP_CLIENT_ID}"
+
+        url = f"https://login.microsoftonline.com/{config.AUTH_TENANT_ID}/oauth2/token"
+        response = await client.post(url, headers=headers, data=payload)
+        token = response.json()["access_token"]
```
Comment (marrobi) [inline e2e_tests/conftest.py:47]: @LizaShak I'm getting "invalid key" for `access_token` here: https://github.com/microsoft/AzureTRE/runs/4301934706?check_suite_focus=true

Any ideas?

Comment (marrobi) [reply e2e_tests/conftest.py:47]: Sorted, issue with `client_id` in above command - was set to the workspace app ID. Also app registrations needed scope adding and admin consent granting.

**Review by marrobi — APPROVED** [review]
Happy with this.

**Review by eladiw — APPROVED** [review]

**Pull request merged**